packages/material-ui/src/MenuList/MenuList.js

```diff
@@ -105,56 +134,52 @@ class MenuList extends React.Component {
     const currentFocusIndex = items.indexOf(currentFocus);
 
     if (currentFocusIndex !== -1) {
-      return this.setTabIndex(currentFocusIndex);
+      return setCurrentTabIndex(currentFocusIndex);
     }
 
-    if (this.selectedItemRef) {
-      return this.setTabIndex(items.indexOf(this.selectedItemRef));
+    if (selectedItemRef.current) {
+      return setCurrentTabIndex(items.indexOf(selectedItemRef.current));
     }
 
-    return this.setTabIndex(0);
+    return setCurrentTabIndex(0);
   }
 
-  render() {
-    const { children, className, onBlur, onKeyDown, disableListWrap, ...other } = this.props;
-
-    return (
-      <List
-        role="menu"
-        ref={ref => {
-          this.listRef = ReactDOM.findDOMNode(ref);
-        }}
-        className={className}
-        onKeyDown={this.handleKeyDown}
-        onBlur={this.handleBlur}
-        {...other}
-      >
-        {React.Children.map(children, (child, index) => {
-          if (!React.isValidElement(child)) {
-            return null;
-          }
-
-          warning(
-            child.type !== React.Fragment,
-            [
-              "Material-UI: the MenuList component doesn't accept a Fragment as a child.",
-              'Consider providing an array instead.',
-            ].join('\n'),
-          );
-
-          return React.cloneElement(child, {
-            tabIndex: index === this.state.currentTabIndex ? 0 : -1,
-            ref: child.props.selected
-              ? ref => {
-                  this.selectedItemRef = ReactDOM.findDOMNode(ref);
-                }
-              : undefined,
-            onFocus: this.handleItemFocus,
-          });
-        })}
-      </List>
-    );
-  }
+  return (
+    <List
+      role="menu"
+      ref={ref => {
+        listRef.current = ReactDOM.findDOMNode(ref);
+      }}
+      className={className}
+      onKeyDown={handleKeyDown}
+      onBlur={handleBlur}
+      {...other}
+    >
+      {React.Children.map(children, (child, index) => {
+        if (!React.isValidElement(child)) {
+          return null;
+        }
+
+        warning(
+          child.type !== React.Fragment,
+          [
+            "Material-UI: the MenuList component doesn't accept a Fragment as a child.",
+            'Consider providing an array instead.',
+          ].join('\n'),
+        );
+
+        return React.cloneElement(child, {
+          tabIndex: index === currentTabIndex ? 0 : -1,
```

Comment:
Do we need to change the tab index? Could we keep -1 (menu), 0 (dropdown) or nothing (comboxbox) depending on the mode? If we might be able to can kill the `currentTabIndex` state.

Reply:
I intend on tackling that next, but I want to take some time to layout all the different scenarios related to tab index in a separate issue and make sure I know what we want the behavior to be in each. It isn't clear to me yet whether or not the dropdown case can be that simple.